native/acter/src/api/message.rs

```diff
@@ -291,10 +295,207 @@ impl RoomEventItem {
+                                Some(old) => MsgContent::from_text(format!("{} → {}", old, new)),
+                                None => MsgContent::from_text(format!("{}", new)),
+                            },
+                            (None, Some(prev)) => match &prev.alias {
+                                Some(old) => MsgContent::from_text(format!("{}", old)),
+                                None => MsgContent::from_text("canonical alias unset".to_string()),
+                            },
+                            (Some(new), None) => MsgContent::from_text(format!("{}", new)),
+                            (None, None) => {
+                                MsgContent::from_text("canonical alias unset".to_string())
+                            }
+                        };
+                        me.msg_content(Some(msg_content));
+                    }
+                }
+                if let AnyOtherFullStateEventContent::SpaceParent(c) = s.content() {
+                    if let FullStateEventContent::Original {
+                        content,
+                        prev_content,
+                    } = c
+                    {
+                        let msg_content = match (content.canonical, &prev_content) {
+                            (new, Some(prev)) => match prev.canonical {
+                                old => MsgContent::from_text(format!("{:?} → {:?}", old, new)),
+                            },
+                            (new, None) => MsgContent::from_text(format!("{}", new)),
+                        };
+                        me.msg_content(Some(msg_content));
+                    }
+                }
+
+                if let AnyOtherFullStateEventContent::SpaceChild(c) = s.content() {
+                    if let FullStateEventContent::Original {
+                        content,
+                        prev_content,
+                    } = c
+                    {
+                        let msg_content = match (content.suggested, &prev_content) {
+                            (new, Some(prev)) => match prev.suggested {
+                                old => MsgContent::from_text(format!("{} → {}", old, new)),
+                            },
+                            (new, None) => MsgContent::from_text(format!("{}", new)),
+                        };
+                        me.msg_content(Some(msg_content));
+                    }
+                }
+
+                if let AnyOtherFullStateEventContent::RoomEncryption(c) = s.content() {
+                    if let FullStateEventContent::Original {
+                        content,
+                        prev_content,
+                    } = c
+                    {
+                        let msg_content = match (&content.algorithm, &prev_content) {
+                            (new, Some(prev)) => match &prev.algorithm {
+                                old => MsgContent::from_text(format!("{:?} → {:?}", old, new)),
```

Comment:
why use the debug notation here?
```suggestion
                                old => MsgContent::from_text(format!("{old} → {new}")),
```